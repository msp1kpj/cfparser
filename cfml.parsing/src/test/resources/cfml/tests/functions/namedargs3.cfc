foo.bar("argNameOne" = argValueOne, "argNameTwo" = argValueTwo);