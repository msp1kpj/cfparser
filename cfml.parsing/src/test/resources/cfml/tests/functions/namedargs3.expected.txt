/*===TOKENS===*/
IDENTIFIER               <foo>
'.'                      <.>
IDENTIFIER               <bar>
'('                      <(>
OPEN_STRING              <">
STRING_LITERAL           <argNameOne>
CLOSE_STRING             <">
'='                      <=>
IDENTIFIER               <argValueOne>
','                      <,>
OPEN_STRING              <">
STRING_LITERAL           <argNameTwo>
CLOSE_STRING             <">
'='                      <=>
IDENTIFIER               <argValueTwo>
')'                      <)>
';'                      <;>
/*===TREE===*/
(scriptBlock
  (element
    (statement
      (startExpression
        (compareExpression
          (baseExpression
            (unaryExpression
              (memberExpression
                (identifier foo)
                .
                (functionCall
                  (identifierOrReservedWord (identifier bar))
                  (
                  (argumentList
                    (argument
                      (argumentName (stringLiteral " (stringLiteralPart argNameOne) "))
                      =
                      (startExpression
                        (compareExpression
                          (baseExpression (unaryExpression (memberExpression (identifier argValueOne))))
                        )
                      )
                    )
                    ,
                    (argument
                      (argumentName (stringLiteral " (stringLiteralPart argNameTwo) "))
                      =
                      (startExpression
                        (compareExpression
                          (baseExpression (unaryExpression (memberExpression (identifier argValueTwo))))
                        )
                      )
                    )
                  )
                  )
                )
              )
            )
          )
        )
      )
      ;
    )
  )
)
/*======*/
/*===DECOMPILE===*/
foo.bar('argNameOne' = argValueOne, 'argNameTwo' = argValueTwo)
/*======*/